File: Spa-Salon-BackEnd/Domain/Entities/Employee.cs
namespace Spa_Salon_BackEnd.Domain.Entities;

public class Employee
{
    public Guid Id { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Position { get; set; } = string.Empty;
    public DateTime HireDate { get; set; }
    public Guid? UserId { get; set; }

    public User? User { get; set; }
    public ICollection<Service> Services { get; set; } = new List<Service>();
    public ICollection<AppointmentService> AppointmentServices { get; set; } = new List<AppointmentService>();
}

File: Spa-Salon-FrontEnd/src/types/employee.ts
import type { AppointmentStatus } from './common';

export interface Employee {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  position: string;
  hasUserAccount: boolean;
}

export interface EmployeeAppointmentService {
  id: string;
  serviceName: string;
  customerName: string;
  appointmentDateTime: string;
  status: AppointmentStatus;
  price: number;
}

export interface EmployeeDetail extends Employee {
  hireDate: string;
  userId?: string;
  appointmentServices: EmployeeAppointmentService[];
}

export interface CreateEmployee {
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  phone?: string;
  position: string;
  hireDate: string;
}

export interface UpdateEmployee {
  firstName?: string;
  lastName?: string;
  email?: string;
  phone?: string;
  position?: string;
}
